GVRf/Framework/jni/objects/hybrid_object_jni.cpp

```diff
@@ -28,7 +28,9 @@ Java_org_gearvrf_NativeHybridObject_delete(JNIEnv * env,
 JNIEXPORT void JNICALL
 Java_org_gearvrf_NativeHybridObject_delete(JNIEnv * env,
         jobject obj, jlong jhybrid_object) {
-    delete reinterpret_cast<HybridObject*>(jhybrid_object);
+    HybridObject* hybrid_object = reinterpret_cast<HybridObject*>(jhybrid_object);
+    delete hybrid_object;
+    hybrid_object = nullptr;
```

Comment:
OK. Problem found and here is a solution.

The reason why GC will crash with RenderPass class is not because of dangling pointer. The reason is that in Java side, you define GVRRenderPass as a subclass of GVRHybridObject. But in c++ side, RenderPass is not a subclass of HybridObject.

So when the codes reach to NativeHybridObject_delete, it tries to turn any pointer from GVRHybridObject into a HybridObject pointer in C++, but at the moment, RenderPass is actually not a HybridObject pointer. That is why delete will finally lead to a crash. Actually you don't have to do "hybrid_object = nullptr;" . Instead, you just need to set RenderPass in C++ as a subclass of HybridObject and the crash will go away(I tried).